src/vs/platform/environment/node/argv.ts

```diff
@@ -167,37 +194,66 @@ export interface ErrorReporter {
 	onMultipleValues(id: string, usedValue: string): void;
 	onEmptyValue(id: string): void;
 	onDeprecatedOption(deprecatedId: string, message: string): void;
+
+	getSubcommandReporter?(commmand: string): ErrorReporter;
 }
 
-const ignoringReporter: ErrorReporter = {
+const ignoringReporter = {
 	onUnknownOption: () => { },
 	onMultipleValues: () => { },
 	onEmptyValue: () => { },
 	onDeprecatedOption: () => { }
 };
 
 export function parseArgs<T>(args: string[], options: OptionDescriptions<T>, errorReporter: ErrorReporter = ignoringReporter): T {
+	const firstArg = args.find(a => a.length > 0 && a[0] !== '-');
+
 	const alias: { [key: string]: string } = {};
 	const string: string[] = ['_'];
 	const boolean: string[] = [];
```

Comment:
I think using `string` or `boolean` as variable name is confusing. What about `stringArgs` or `booleanArgs`?

Reply:
https://github.com/microsoft/vscode/pull/163550